InteractionBase/src/vis/Menu.cpp

```diff
@@ -186,7 +187,15 @@ bool Menu::sceneEventFilter(QGraphicsItem* watched, QEvent* event)
 		else if (keyEvent->key() == Qt::Key_Return)
 		{
 			event->accept();
-			if (executeFocused())
+			qDebug() << inFocusMode_;
+			if (inFocusMode_)
```

Comment:
So the body of this if statement duplicates code with the previous case of F2. You can avoid all of this if you change the condition for the F2 case to the following `if (keyEvent->key() == Qt::Key_F2 || (inFocusMode_ && (keyEvent->key() == Qt::Key_Enter || Qt::Key_Return))). Then the code here, for Return, doesn't change (except also adding Enter to the condition)